File: backend/src/main/java/com/khoinguyen/orderfood/model/Product.java
package com.khoinguyen.orderfood.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "products")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    @Lob
    private String description;

    @Column(nullable = false)
    private BigDecimal price;
    @Lob
    private String image;
//    @Column(unique = true, nullable = false)
//    private String slug;
    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;
    @Column(nullable = false)
    private boolean available = true;

    @CreationTimestamp
    private LocalDateTime createdAt;
    @UpdateTimestamp
    private LocalDateTime updatedAt;

}

File: frontend/src/types/Product.ts
import { ICategory } from "./Category";

export type IProduct = {
  id: number;
  name: string;
  image: string;
  description: string;
  price: number;
  available: boolean;
  createdAt: string;
  category: ICategory;
};
export type IProductOrder = {
  id: number;
  name: string;
  image: string;
  price: number;
  message: string;
  quantity: number;
  totalPrice: number;
};
